src/jit/gentree.cpp

```diff
@@ -505,6 +525,74 @@ bool GenTree::IsNodeProperlySized() const
 
 /*****************************************************************************/
 
+#if MEASURE_NODE_SIZE
+
+void GenTree::DumpNodeSizes(FILE* fp)
+{
+    // Dump the sizes of the various GenTree flavors
+
+#if SMALL_TREE_NODES
+    fprintf(fp, "Small tree node size = %3u bytes\n", TREE_NODE_SZ_SMALL);
+#endif // SMALL_TREE_NODES
+    fprintf(fp, "Large tree node size = %3u bytes\n", TREE_NODE_SZ_LARGE);
+    fprintf(fp, "\n");
+
+#if SMALL_TREE_NODES
+
+    // Verify that node sizes are set kosherly and dump sizes
+    for (unsigned op = GT_NONE+1; op < GT_COUNT; op++)
+    {
+        unsigned    needSize = s_gtTrueSizes[op];
+        unsigned    nodeSize = s_gtNodeSizes[op];
+
+        const char* structNm = OpStructName((genTreeOps)op);
+        const char* operName =       OpName((genTreeOps)op);
+
+        // Have we seen this struct flavor before?
+        for (unsigned mop = GT_NONE+1; mop < op; mop++)
+        {
+            if (strcmp(structNm, OpStructName((genTreeOps)mop)) == 0)
+            {
+                structNm = nullptr;
+                break;
+            }
+        }
+
+        if (structNm != nullptr)
```

Comment:
It just occurred to me that a "later" error may in fact be useful - the small / large size for multiple operators that use the same GenTree flavor could be set differently, and only the second / later operator(s) may have SMALL in s_gtNodeSizes[] so an error should be printed for those later operators but not the first. Very unlikely but I think it could happen. Well none of this should ever happen but YNWIM.